frontend/src/scenes/pipeline/PipelineNodeNew.tsx

```diff
@@ -80,28 +97,52 @@ export function PipelineNodeNew(
         return <>Plugin ID {pluginId}</>
     }
     if (batchExportDestination) {
+        if (stage !== PipelineStage.Destination) {
+            return <NotFound object={batchExportDestination} />
+        }
         return <>Batch Export Destination {batchExportDestination}</>
     }
 
     if (stage === PipelineStage.Transformation) {
-        // Show a list of transformations
-        const { plugins, loading } = useValues(pipelineTransformationsLogic)
-        const targets = Object.values(plugins).map(convertPluginToTableEntry)
-        return nodeOptionsTable(stage, targets, loading)
+        return <TransformationOptionsTable />
     } else if (stage === PipelineStage.Destination) {
-        const { plugins, loading } = useValues(pipelineDestinationsLogic)
-        // Show a list of destinations - TODO: add batch export destinations too
-        const targets = Object.values(plugins).map(convertPluginToTableEntry)
-        return nodeOptionsTable(stage, targets, loading)
+        return <DestinationOptionsTable />
     } else if (stage === PipelineStage.SiteApp) {
-        const { plugins, loading } = useValues(frontendAppsLogic)
-        const targets = Object.values(plugins).map(convertPluginToTableEntry)
-        return nodeOptionsTable(stage, targets, loading)
+        return <SiteAppOptionsTable />
     }
-    return <>Creation is unavailable for {stage}</>
+    return <NotFound object="pipeline new options" />
+}
+
+function TransformationOptionsTable(): JSX.Element {
+    const { plugins, loading } = useValues(pipelineTransformationsLogic)
+    const targets = Object.values(plugins).map(convertPluginToTableEntry)
+    return <NodeOptionsTable stage={PipelineStage.Transformation} targets={targets} loading={loading} />
 }
 
-function nodeOptionsTable(stage: PipelineStage, targets: TableEntry[], loading: boolean): JSX.Element {
+function DestinationOptionsTable(): JSX.Element {
+    const { batchExportServiceNames } = useValues(pipelineNodeNewLogic)
+    const { plugins, loading } = useValues(pipelineDestinationsLogic)
+    const pluginTargets = Object.values(plugins).map(convertPluginToTableEntry)
+    const batchExportTargets = Object.values(batchExportServiceNames).map(convertBatchExportToTableEntry)
+    const targets = [...batchExportTargets, ...pluginTargets]
+    return <NodeOptionsTable stage={PipelineStage.Destination} targets={targets} loading={loading} />
+}
+
+function SiteAppOptionsTable(): JSX.Element {
+    const { plugins, loading } = useValues(frontendAppsLogic)
+    const targets = Object.values(plugins).map(convertPluginToTableEntry)
+    return <NodeOptionsTable stage={PipelineStage.SiteApp} targets={targets} loading={loading} />
```

Comment:
I assume its just in progress issues, but locally transformations is empty for me and site apps includes one app "automatic cohorts creator"

Reply:
likely because you don't have any apps installed, this part seemed to have worked well for me and we can verify in prod.